contrib/dracut/90zfs/zfs-lib.sh.in

```diff
@@ -66,6 +66,46 @@ import_pool() {
             warn "ZFS: Unable to import pool ${pool}"
             return 1
         fi
+
+	# Set elevator=none or noop on the root pool's vdevs' disks.  ZFS
+	# already does this for whole_disk vdevs (for all pools), so this is
+	# only important for partitions.
+	if getargbool 0 zfs_elevator ; then
+		zpool status -L "${pool}" 2> /dev/null |
+		    awk '/^\t / && !/(mirror|raidz)/ {
+			dev=$1;
+			sub(/[0-9]+$/, "", dev);
+			print dev
+		    }' |
```

Comment:
I'm personally more partial to this to get the list of blockdevs.

    zpool list -HLpv "${pool}" | grep -Pv '(^\t+(mirror|raidz)|^[[:alnum:]])' | awk '{ print $1 }'

The problem with both the `zpool status` and `zpool list` is that it relies on /dev/mapper being links (one host has it, the other doesn't...).  Found a way around it with lsblk though.

    for blk in \
        $(zpool list -HLpv node02-zp00 | \
        grep -Pv '(^\t+(mirror|raidz)|^[[:alnum:]])' | \
        awk '{ print $1 }'); do \
    lsblk -o name,kname -l | grep $blk | awk '{print $2}';
    done

Reply:
the lsblk line needs to check the right field in the grep though (and fall back to what zfs reported).

```bash
for blk in $(zpool list -HLpv node02-zp00 | grep -Pv '(^\t+(mirror|raidz)|^[[:alnum:]])' | awk '{ print $1 }'); do
    lsblk -o name,kname -l | grep "^$blk" | awk '{print $2}' || echo "${blk}";
done
```

Reply:
it's been said that we don't care about device-mappings, so ok...